src/main/scala/org/ergoplatform/wallet/secrets/JsonSecretStorage.scala

```diff
@@ -0,0 +1,80 @@
+package org.ergoplatform.wallet.secrets
+
+import java.io.{File, PrintWriter}
+import java.util.UUID
+
+import io.circe.generic.auto._
+import io.circe.parser._
+import io.circe.syntax._
+import org.ergoplatform.wallet.crypto
+import org.ergoplatform.wallet.settings.{EncryptionSettings, WalletSettings}
+import scorex.util.encode.Base16
+
+import scala.util.Try
+
+/**
+  * Secret storage backend. Stores encrypted seed in json file in the file system.
+  * Responsible for managing access to the secrets.
+  */
+final class JsonSecretStorage(val secretFile: File, encryptionSettings: EncryptionSettings)
+  extends SecretStorage {
+
+  private var unlockedSecrets: Map[Int, SecureSecret] = Map.empty
+
+  override def isLocked: Boolean = unlockedSecrets.isEmpty
+
+  override def secrets: Map[Int, SecureSecret] =
```

Comment:
@catena2w, `unlockedSecrets` is mutable `var` which is made private in order to avoid mutation from outside the class. `def secrets` is reader method.